File: widget/src/main/java/ru/kest/trainswidget/model/yandex/ScheduleResponse.java
package ru.kest.trainswidget.model.yandex;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.Date;
import java.util.List;

/**
 * Created by KKharitonov on 06.01.2016.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@Data
public class ScheduleResponse {

    private List<TrainThread> threads;

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class TrainThread {

        private Date arrival;
        private Date departure;
        private float duration;

        private Thread thread;

        @Data
        @JsonIgnoreProperties(ignoreUnknown = true)
        public static class Thread {

            private String shortTitle;

        }
    }

}

File: widget/src/main/java/ru/kest/trainswidget/model/yandex/ScheduleResponse.kt
package ru.kest.trainswidget.model.yandex

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.util.*

/**
 * Yandex API DTO (for convert from json)
 *
 * Created by KKharitonov on 06.01.2016.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
data class ScheduleResponse(val threads: List<ApiTrainThread> = emptyList()) {

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ApiTrainThread(
            val arrival: Date? = null,
            val departure: Date? = null,
            val duration: Float = 0.toFloat(),
            val thread: Thread? = null
    ) {

        @JsonIgnoreProperties(ignoreUnknown = true)
        data class Thread(val shortTitle: String = "")
    }
}
